packages/rainbowkit/src/wallets/walletConnectors/coinbaseWallet/coinbaseWallet.ts

```diff
@@ -106,3 +107,10 @@ export const coinbaseWallet = ({
     },
   };
 };
+
+export const coinbaseWallet = Object.assign<
+  typeof wallet,
+  { preference: CoinbaseWalletParameters<'4'>['preference'] }
```

Comment:
This would give us cleaner typescript builds.

Here is how `coinbaseWallet.d.ts` build file would look like:

```
import { CoinbaseWalletParameters } from 'wagmi/connectors';
import { Wallet } from '../../Wallet';
export interface CoinbaseWalletOptions {
    appName: string;
    appIcon?: string;
}
export declare const coinbaseWallet: (({ appName, appIcon }: CoinbaseWalletOptions) => Wallet) & {
    preference: CoinbaseWalletParameters<'4'>['preference'];
};
```

Reply:
Was there a reason we shifted away from the Interface type? I lean in that direction because the `preference` would be undefined unless the user specifies.